### ISSUE TITLE
docker image: add option to disable remco for use a volume with existing config files

### ISSUE THREAD

**Issue by dginhoux** [rundeck/rundeck#6334]
Hi everyone, 

In my work we provide one rundeck per customer.
I'm trying to replace with docker an old model "on customer, one debian vm, rundeck oss installed".

Is there a way to disable remco and use existing config file ?
Because every customers have their conf files, with lot of différences.

I've tried in the stack yml to put some volume (bind) as : 

```
   volumes:
      - ./app-conf/framework.properties:/home/rundeck/etc/framework.properties:ro
      - ./app-conf/realm.properties:/home/rundeck/server/config/realm.properties:ro
      - ./app-conf/rundeck-config.properties:/home/rundeck/server/config/rundeck-config.properties:ro
```

But many few problems, because after read the entry point, it fail because it overwrite my files and failed because read only...

**Comment by ProTip** [comment]
Hi @Dany21000 ,

There is not currently however this is a good enhancement request. Right now you would need to customize the entrypoint.

**Comment by Dany21000** [comment]
Yes, i'm working on it, i'll create a volume bind to my own entrypoint.

**Comment by cloudlady911** [comment]
I was able to work around this issue by doing the following:
My problem was that I wanted to change the  `rundeck.api.tokens.duration.max` option.  If I changed it in my config file, when the docker container restarts that setting got overwritten.

I found that my settings were getting overwritten by the settings in this file: /etc/remco/templates/rundeck-config.properties.  Specifically this line:
 
`rundeck.api.tokens.duration.max={{ getv("/rundeck/api/tokens/duration/max", "30d") }}`

So I created a file in my docker build directory named rundeck-config.properties and copied all the settings from the file in rundeck.  Then I modified the above setting to: 

`rundeck.api.tokens.duration.max={{ getv("/rundeck/api/tokens/duration/max", "0d") }}`

Then I added a line in my Dockerfile to copy the file during the build:

`COPY rundeck-config.properties /etc/remco/templates/rundeck-config.properties`

I built my image and after the new rundeck container came up I looked at the rundeck-config.properties file in /:/home/rundeck/server/config folder and the setting now says rundeck.api.tokens.duration.max=0d.

Now I am able to create API tokens that do not expire.

This is really kind of a hack and should not be this difficult.  If an application is going to be run in docker it should support passing settings like this through the environment variables.

**Comment by fdevans** [comment]
Thanks for the report. As part of a cleanup of our issue tracker, we're closing issues older than 12 months, in line with Rundeck's support policy of one year from a release's date (see the [release calendar](https://docs.rundeck.com/docs/history/release-calendar.html) for currently supported versions). This isn't a judgment that the report is invalid — many of these are likely still real — but we can't verify against a version this old.

If you're still seeing this on a currently supported release (6.1.x or 5.20.x+), please open a new issue using our [bug report template](https://github.com/rundeck/rundeck/issues/new?template=bug_report.yml) with your version and reproduction steps. Please don't reopen this issue — a fresh issue keeps things tracked against a supported version and gets it in front of us properly. We're actively working on improving how issues get submitted and triaged going forward.

Thanks for your patience,
The Rundeck Team
